metabase/metabase
# Add chart color generation

Epic https://github.com/metabase/metabase/issues/22816

Changes:
- Adds a way to automatically fill in missing chart colors
- Changes the palette preview to show 3 bars with different tints/shades combinations
- Adjusts the way how tints and shades are generated to prevent generating colors that are too light

How to test:
- Run Metabase EE
- Go to Admin > Settings > Appearance
- Reset all charts colors to default and click `Generate chart colors`. It should set all main accents.
- Reset all charts colors to default and fill in some of them, but not all. Clicking on `Generate` should set missing values based on the first not empty color.
- Make sure that after generating colors all manually set tints (light variants) and shades (dark variants) are cleared out.

<img width="1369" alt="Screenshot 2022-06-03 at 17 04 59" src="https://user-images.githubusercontent.com/8542534/171871954-e92ac769-60f0-4ce6-998c-4b7f6c7632c6.png">
